src/main/java/com/chatapp/vartalaab/controller/MessageController.java

```diff
@@ -0,0 +1,28 @@
+package com.chatapp.vartalaab.controller;
+
+import com.chatapp.vartalaab.document.Message;
+import com.chatapp.vartalaab.repository.MongoMessageRepository;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@Slf4j
+public class MessageController {
+    MongoMessageRepository mongoMessageRepository;
+
+    MessageController(MongoMessageRepository mongoMessageRepository){
+        this.mongoMessageRepository=mongoMessageRepository;
+    }
+
+    //fetches messages based on both chatID and UserID
+    @GetMapping("/getMessages/{chatId}")
+    public ResponseEntity<?> getMessages(@PathVariable String chatId){
```

Comment:
//fetches messages based on both chatID and UserID  ? is this an old comment that you intended to remove ?

Reply:
No, it indeed works for both chatId and userId, since chatId is made out of two userIds and method in corresponding interface works based on regex and can identify both.